src/lib/expression/evaluation/expression_evaluator.cpp

```diff
@@ -0,0 +1,1054 @@
+    } else {
+      pmr_concurrent_vector<ColumnDataType> values(_output_row_count);
+
+      for (auto chunk_offset = ChunkOffset{0}; chunk_offset < _output_row_count; ++chunk_offset) {
+        values[chunk_offset] = view.value(chunk_offset);
+      }
+
+      if (view.is_nullable()) {
+        pmr_concurrent_vector<bool> nulls(_output_row_count);
+        for (auto chunk_offset = ChunkOffset{0}; chunk_offset < _output_row_count; ++chunk_offset) {
+          nulls[chunk_offset] = view.null(chunk_offset);
+        }
+        column = std::make_shared<ValueColumn<ColumnDataType>>(std::move(values), std::move(nulls));
+      } else {
+        column = std::make_shared<ValueColumn<ColumnDataType>>(std::move(values));
+      }
+    }
+    // clang-format on
+  });
+
+  return column;
+}
+
+template <>
+std::shared_ptr<ExpressionResult<int32_t>> ExpressionEvaluator::_evaluate_logical_expression<int32_t>(
+    const LogicalExpression& expression) {
+  const auto& left = *expression.left_operand();
+  const auto& right = *expression.right_operand();
+
+  // clang-format off
+  switch (expression.logical_operator) {
+    case LogicalOperator::Or:  return _evaluate_binary_with_custom_null_logic<int32_t, TernaryOr>(left, right);
+    case LogicalOperator::And: return _evaluate_binary_with_custom_null_logic<int32_t, TernaryAnd>(left, right);
+  }
+  // clang-format on
+}
+
+template <typename R>
+std::shared_ptr<ExpressionResult<R>> ExpressionEvaluator::_evaluate_logical_expression(
+    const LogicalExpression& expression) {
+  Fail("LogicalExpression can only output int32_t");
+}
+
+template <typename R, typename Functor>
+std::shared_ptr<ExpressionResult<R>> ExpressionEvaluator::_evaluate_binary_with_default_null_logic(
+    const AbstractExpression& left_expression, const AbstractExpression& right_expression) {
+  auto result = std::shared_ptr<ExpressionResult<R>>{};
+
+  _resolve_to_expression_results(left_expression, right_expression, [&](const auto& left, const auto& right) {
+    using LeftDataType = typename std::decay_t<decltype(left)>::Type;
+    using RightDataType = typename std::decay_t<decltype(right)>::Type;
+
+    if constexpr (Functor::template supports<R, LeftDataType, RightDataType>::value) {
+      const auto result_size = _result_size(left.size(), right.size());
+      auto nulls = _evaluate_default_null_logic(left.nulls, right.nulls);
+
+      // Using three different branches instead of views, which would generate 9 cases.
+      std::vector<R> values(result_size);
+      if (left.is_literal() == right.is_literal()) {
+        for (auto row_idx = ChunkOffset{0}; row_idx < result_size; ++row_idx) {
+          Functor{}(values[row_idx], left.values[row_idx], right.values[row_idx]);
+        }
+      } else if (right.is_literal()) {
+        for (auto row_idx = ChunkOffset{0}; row_idx < result_size; ++row_idx) {
+          Functor{}(values[row_idx], left.values[row_idx], right.values.front());
```

Comment:
No idea where to put this, but if you ignore null values, don't you have to deal with div/0?

```
(debug)> load src/test/tables/int_float_with_null.tbl tbl2
Loading src/test/tables/int_float_with_null.tbl into table "tbl2" ...
(debug)> SELECT a/(a-a) FROM tbl2
Floating point exception: 8
```

Reply:
Div by zero is NULL now, as it is in SQLite and MySQL. That was broken at the time of you writing, sorry.